src/Data/Functor/Contravariant/Divisible.hs

```diff
@@ -603,3 +607,58 @@ instance Decidable SettableStateVar where
 -- In addition, we expect the same kind of distributive law as is satisfied by the usual
 -- covariant 'Alternative', w.r.t 'Applicative', which should be fully formulated and
 -- added here at some point!
+
+-- | Analagous to `(*>)`
+--
+-- @
+-- showing :: 'Show' a => 'Op' 'String' a
+-- showing = 'Op' 'show'
+--
+-- string :: String -> 'Op' 'String' ()
+-- string = 'Op' '.' 'const'
+--
+-- greeting :: Show a => 'Op' 'String' a
+-- greeting = string "Hello " '>*' showing
+-- @
+(>*) :: Divisible f => f () -> f a -> f a
+before >* p = ((),) >$< before >*< p
+
+infixr 5 >*
+
+-- | Analagous to `(<*)`
+--
+-- @
+-- emphatic :: 'Op' 'String' a -> 'Op' 'String' a
+-- emphatic opstring = opstring '*<' string "!"
+-- @
+(*<) :: Divisible f => f a -> f () -> f a
+p *< after = (,()) >$< p >*< after
+
+infixr 5 *<
+
+mayhaps :: Maybe a -> Either () a
+mayhaps m = case m of
+  Nothing -> Left ()
+  Just a -> Right a
+
+-- | Zero or one.
+optionalD :: Decidable f => f a -> f (Maybe a)
+optionalD = choose mayhaps conquered
+
+-- | Zero or more.
+manyD :: Decidable f => f a -> f [a]
+manyD p = choose (mayhaps . uncons) conquered (many1D p)
+
+-- | One or more.
+many1D :: Decidable f => f a -> f (a,[a])
```

Comment:
It's possibly more ergonomic to work with a `NonEmpty a` than an `(a, [a])`?

Reply:
Maybe, but I don't really think `NonEmpty` is much better (and it has an evil `IsList` instance) than the pair and this matches better with `divide`. I could be convinced otherwise.

Reply:
I don't know either; just wanted to flag it.

Reply:
If we used `NonEmpty a` in `many1D` then you'd have to pay the cost of conjugating by the isomorphism `NonEmpty a <-> (a,[a])` in `many1` (at least if we do the naive thing).

Reply:
I do like NonEmpty here, even if there is a little bit of overhead.

Reply:
Ok, I added a [commit](https://github.com/ekmett/contravariant/pull/71/commits/99dd4e4f39813f70bd3cb2d55d35aa3666386a19) to use `NonEmpty`. I think the cost is `n-1` calls conjugating by the isomorphism for a list of length `n`. Happy to make any other changes to fixity or naming.

Reply:
Or I could just write a helper function and not be naive. Ok, now the cost is just `1` call conjugating by the isomorphism.